pennylane/decomposition/resources.py

```diff
@@ -137,7 +137,9 @@ def __repr__(self):
 
 def _make_hashable(d):
     if isinstance(d, dict):
-        return tuple(sorted(((k, _make_hashable(v)) for k, v in d.items()), key=lambda x: x[0]))
+        return tuple(
+            sorted(((str(k), _make_hashable(v)) for k, v in d.items()), key=lambda x: x[0])
```

Comment:
I think this is fine for now, but in general I want to flag that the order of the factors in the PauliWord is not fixed. 

We don't sort or order them in the `__str__` method of the PauliWord class. This could lead to potentially matching hashes or mismatched hashes when they should be matching. Although I would have to think a bit more before I can tell you for certain if this is going to break something.